packages/core/status/core-status-server-internal/src/status_service.ts

```diff
@@ -184,6 +185,12 @@ export class StatusService implements CoreService<InternalStatusServiceSetup> {
     getOverallStatusChanges(this.overall$, this.stop$).subscribe((message) => {
       this.logger.info(message);
     });
+
+    getPluginsStatusChanges(this.pluginsStatus.getAll$(), this.stop$).subscribe((statusChanges) => {
+      getPluginStatusChangesMessages(statusChanges).forEach((message) => {
+        this.logger.info(message);
+      });
+    });
```

Comment:
This is the entry point to the PR. We start observing plugins statuses and logging meaningful information whenever there are changes.

Reply:
Very cool, overall look good so far!

---

What do you think about combining this into a single, multiline log?

Or, if we're worried about breaking some monitoring regexes that are looking for a status change message maybe keep the single message and then have the "digest" message second (so max two messages)?

My concern is potential interleaving of messages.

Just a thought, need to still do rest of review

Reply:
Good point!

IMO on one hand, some of the messages can be logged independently; they just appear together because the _Observables_ we observe are debouncing events internally.

On the other hand, there are others, like details of each plugin status change, which do belong together.
I have updated these in https://github.com/elastic/kibana/pull/168207/commits/320933d74efa7187168549671f0ebc268547edbf to create some multiline logs.